src/main/java/com/hubspot/maven/plugins/prettier/diff/DefaultDiffGenerator.java

```diff
@@ -0,0 +1,83 @@
+package com.hubspot.maven.plugins.prettier.diff;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.maven.plugin.MojoExecutionException;
+
+public class DefaultDiffGenerator implements DiffGenerator {
+
+  @Override
+  public void generateDiffs(GenerateDiffArgs args) throws MojoExecutionException {
```

Comment:
Should we have a `protected` method that can be reused by subclasses? There seems to be a lot of duplicate code between this and our internal implementation

Reply:
I can try moving all the `Process` stuff to the interface